src/core/common/debug.hpp

```diff
@@ -108,4 +108,19 @@
         }                        \
     } while (false)
 
+/**
+ * @def AssertPointerIsNotNull
+ *
+ * This macro asserts that a given pointer (API input parameter) is not `nullptr`. This macro checks the pointer only
+ * when `OPENTHREAD_CONFIG_ASSERT_CHECK_API_POINTER_PARAM_FOR_NULL` is enabled. Otherwise it is an empty macro.
+ *
+ * @param[in]  aPointer   The pointer variable (API input parameter) to check.
+ *
+ */
+#if OPENTHREAD_CONFIG_ASSERT_CHECK_API_POINTER_PARAM_FOR_NULL
+#define AssertPointerIsNotNull(aPointer) OT_ASSERT((aPointer) != nullptr)
```

Comment:
Is it a good idea to enable it for posix?

Reply:
Good idea. Added it in `openthread-core-posix-config.h`